source/FreeRTOS_IPv6_Utils.c

```diff
@@ -161,6 +172,117 @@ BaseType_t prvChecksumICMPv6Checks( size_t uxBufferLength,
 }
 /*-----------------------------------------------------------*/
 
+/**
+ * @brief Get total length of all extension headers in IPv6 packet.
+ *
+ * @param[in] pucEthernetBuffer The buffer containing the packet.
+ * @param[in] uxBufferLength The number of bytes to be sent or received.
+ * @param[out] pucProtocol The L4 protocol, such as TCP/UDP/ICMPv6.
+ *
+ * @return The total length of all extension headers, or whole buffer length when error detected.
+ */
+size_t usGetExtensionHeaderLength( const uint8_t * pucEthernetBuffer,
+                                   size_t uxBufferLength,
+                                   uint8_t * pucProtocol )
+{
+    uint8_t ucCurrentHeader;
+    const IPPacket_IPv6_t * pxIPPacket_IPv6;
+    uint8_t ucNextHeader = 0U;
+    size_t uxIndex = ipSIZE_OF_ETH_HEADER + ipSIZE_OF_IPv6_HEADER;
+    size_t uxHopSize = 0U;
+    BaseType_t xCurrentOrder = 0;
+    BaseType_t xNextOrder = 0;
+    size_t uxReturn = uxBufferLength;
+
+    if( ( pucEthernetBuffer != NULL ) && ( pucProtocol != NULL ) )
+    {
+        /* MISRA Ref 11.3.1 [Misaligned access] */
+        /* More details at: https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/MISRA.md#rule-113 */
+        /* coverity[misra_c_2012_rule_11_3_violation] */
+        pxIPPacket_IPv6 = ( ( const IPPacket_IPv6_t * ) pucEthernetBuffer );
+        ucCurrentHeader = pxIPPacket_IPv6->xIPHeader.ucNextHeader;
+
+        /* Check if packet has extension header. */
+        if( xGetExtensionOrder( ucCurrentHeader, 0U ) > 0 )
+        {
+            while( ( uxIndex + 8U ) < uxBufferLength )
+            {
+                ucNextHeader = pucEthernetBuffer[ uxIndex ];
+
+                xCurrentOrder = xGetExtensionOrder( ucCurrentHeader, ucNextHeader );
+
+                /* To avoid compile warning if debug print is disabled. */
+                ( void ) xCurrentOrder;
```

Comment:
This line may not be needed as `xCurrentOrder` is getting used in line 277.

Reply:
No, line 277 is using `ucCurrentHeader`, not `xCurrentOrder`.